### PR TITLE
feat(RTL): better RTL support

### PR DESCRIPTION
**Pull request by AmitMY**
#### Short description of what this resolves:
- Much better RTL support

#### Changes proposed in this pull request:
- Add `text-start` and `text-end`. Work exactly the same on `ltr`, but flips directions on `rtl`.
This also avoids confusion, when doing `text-left` on `rtl` it means left, but `text-start` means right
- Add `pull-right` and `pull-left` (bootstrap style naming) for `float`, and flipped on RTL
- Flip every icon except for logos and specific icons that are the same in RTL. (I am from Israel, I made the list)
- Flip tabs border-radius - fixes
image
- Flip `icon-left` and `icon-right` padding

**Ionic Version**: 3.x

**Fixes**: 
Partially: https://github.com/driftyco/ionic/issues/11115
Fully: https://github.com/driftyco/ionic/issues/10685

### CODE REVIEW

**Comment by AmitMY** [comment]
Can I get any comment on this? Am I doing something wrong? is this to be merged sometime, should I change anything?

File: src/components/app/app.scss [hunk 1]
```diff
@@ -332,3 +339,24 @@ ion-footer {
 [text-capitalize] {
   text-transform: capitalize;
 }
+
+// Float
+// --------------------------------------------------
+
+[pull-left],
```
Comment (brandyscarney) [inline src/components/app/app.scss]: Could we do `pull-left`, `pull-right`, `pull-start`, and `pull-end` here and make them work like the `text-*` attributes do? Or is there a reason not to?

File: src/components/button/button-icon.scss [hunk 2]
```diff
@@ -11,18 +11,34 @@
...
 }
 
+[dir="rtl"] [icon-left] ion-icon {
+  @include button-icon;
+
+  padding-left: .4em;
+  padding-right: initial;
+}
+
+[dir="rtl"] [icon-right] ion-icon {
```
Comment (brandyscarney) [inline src/components/button/button-icon.scss]: The same question as above with `icon-*`.

File: src/fonts/ionicons.scss [hunk 3]
```diff
@@ -8,6 +8,7 @@ $ionicons-font-path: $font-path !default;
 
 @import "ionicons-icons";
 @import "ionicons-variables";
+@import "~hail2u-scss-functions/string/_str-replace";
```
Comment (brandyscarney) [inline src/fonts/ionicons.scss]: I'm not sure about this for icons. I wonder if there is a better way to decide if the icon should or shouldn't be flipped on rtl. We're thinking maybe we could add some metadata to the icon itself and look at it that way? We need to discuss this more but maybe this could be split out from this PR for further discussion.

File: src/components/app/app.scss (hunk 1 above)
Comment (AmitMY) [inline src/components/app/app.scss]: Done, moved to a separate PR: https://github.com/driftyco/ionic/pull/11214

File: src/components/button/button-icon.scss (hunk 2 above)
Comment (AmitMY) [inline src/components/button/button-icon.scss]: That's gonna be a problem.
`icon-left` is for CSS, so when you do:
```
<button ion-button icon-left>
    Button
    <ion-icon name="chatbubbles"></ion-icon>
</button>
```

It doesn't move the icon to the left, just adds some wrong padding.

File: src/fonts/ionicons.scss (hunk 3 above)
Comment (AmitMY) [inline src/fonts/ionicons.scss]: Hmm.. ok
How about for starters, do:
`ion-icon[area-label^="arrow"] { @extends [flip] }` ?
To be honest, most of the icons don't matter, its just the arrows that are going the other way..

**Comment by brandyscarney** [comment]
@AmitMY Yeah could you split this into separate PRs. It's easier to merge them and review them that way. :smile:

**Comment by AmitMY** [comment]
@brandyscarney Thanks.
All 5 PRs created, and are easily mergable (perhaps except for `icon-x` which probably requires further discussion)
